ethereum/contracts/pyth/PythGovernance.sol

```diff
@@ -45,14 +45,17 @@ abstract contract PythGovernance is PythGetters, PythSetters, PythGovernanceInst
         if (gi.action == GovernanceAction.UpgradeContract) {
             require(gi.targetChainId != 0, "upgrade with chain id 0 is not possible");
             upgradeContract(gi.payload);
-        } else if (gi.action == GovernanceAction.SetGovernanceDataSource) {
-            setGovernanceDataSource(gi.payload);
+        } else if (gi.action == GovernanceAction.TransferGovernanceDataSource) {
+            require(gi.targetChainId == 0, "governance data source change should be applied to all chains");
+            transferGovernanceDataSource(gi.payload);
```

Comment:
Minor point, but we could parse the payload before calling the handlers:
```
transferGovernanceDataSource(parseTransferGovernanceDataSourcePayload(gi.payload));
```
which would then allow us to use meaningful types for the handler function signatures.

Reply:
Nice point, I wish you told it in the earlier PR for governance :D

I'll do it in a separate PR as it touches other places too.